code/modules/reagents/reagent_containers/dropper.dm

```diff
@@ -19,110 +18,112 @@
+		if(reagents.total_volume)
+			if(istype(target, /obj/machinery/artifact))
+				reagents.clear_reagents()
+				user << "<span class='notice'>You squirt the solution onto the [target]!</span>"
+				update_icon()
+		return
 
-				var/mob/living/M = target
+	if(reagents.total_volume)
 
-				var/list/injected = list()
-				for(var/datum/reagent/R in src.reagents.reagent_list)
-					injected += R.name
-				var/contained = english_list(injected)
-				M.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has been squirted with [src.name] by [user.name] ([user.ckey]). Reagents: [contained]</font>")
-				user.attack_log += text("\[[time_stamp()]\] <font color='red'>Used the [src.name] to squirt [M.name] ([M.key]). Reagents: [contained]</font>")
-				msg_admin_attack("[user.name] ([user.ckey]) squirted [M.name] ([M.key]) with [src.name]. Reagents: [contained] (INTENT: [uppertext(user.a_intent)]) (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[user.x];Y=[user.y];Z=[user.z]'>JMP</a>)")
-				if(!iscarbon(user))
-					M.LAssailant = null
-				else
-					M.LAssailant = user
+		if(target.reagents.total_volume >= target.reagents.maximum_volume)
+			user << "<span class='warning'>[target] is full.</span>"
+			return
 
-			trans = src.reagents.trans_to(target, amount_per_transfer_from_this)
-			user << "<span class='notice'>You transfer [trans] units of the solution.</span>"
-			if (src.reagents.total_volume<=0)
-				filled = 0
-				icon_state = "dropper[filled]"
+		if(!target.is_open_container() && !ismob(target) && !is_type_in_list(target, injectable_types)) //You can inject humans and food but you cant remove the shit.
+			user << "<span class='warning'>You cannot directly fill this object.</span>"
+			return
 
-			// /vg/: Logging transfers of bad things
-			if(isobj(target))
-				if(istype(reagents_to_log) && reagents_to_log.len && target.log_reagents)
-					var/list/badshit=list()
-					for(var/bad_reagent in reagents_to_log)
-						if(reagents.has_reagent(bad_reagent))
-							badshit += reagents_to_log[bad_reagent]
-					if(badshit.len)
-						var/hl="<span class='danger'>([english_list(badshit)])</span>"
-						message_admins("[user.name] ([user.ckey]) added [trans]U to \a [target] with [src].[hl] (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[user.x];Y=[user.y];Z=[user.z]'>JMP</a>)")
-						log_game("[user.name] ([user.ckey]) added [trans]U to \a [target] with [src].")
+		var/trans = 0
+
+		if(ismob(target))
+			if(istype(target , /mob/living/carbon/human))
+				var/mob/living/carbon/human/victim = target
+
+				var/obj/item/safe_thing = victim.get_body_part_coverage(EYES)
+
+				if(safe_thing)
+					if(!safe_thing.reagents)
+						safe_thing.create_reagents(100)
+					trans = src.reagents.trans_to(safe_thing, amount_per_transfer_from_this)
+
+					for(var/mob/O in viewers(world.view, user))
+						O.show_message(text("<span class='danger'>[] tries to squirt something into []'s eyes, but fails!</span>", user, target), 1)

```

Comment:
I swear I am about one more refactor PR from just going through the code to remove every instance of the garbage so I don't have to fix it over and over.